tfx_addons/huggingface_pusher/executor.py

```diff
@@ -0,0 +1,137 @@
+            This is set to 0 when the input model is not blessed, and it is
+            set to 1 when the model is successfully pushed.
+            - pushed_version : string value to indicate the current model ver
+            sion. This is decided by time.time() Python built-in function.
+            - repo_id : model repository ID where the model is pushed to. This
+            follows the format of f"{username}/{repo_name}".
+            - branch : branch name where the model is pushed to. The branch na
+            me is automatically assigned to the same value of pushed_version.
+            - commit_id : the id from the commit history (branch name could be
+            sufficient to retreive a certain version of the model) of the mo
+            del repository.
+            - repo_url : model repository URL. It is something like f"https://
+            huggingface.co/{repo_id}/{branch}"
+            - space_url : space repository URL. It is something like f"https://
+            huggingface.co/{repo_id}"f
+        exec_properties: An optional dict of execution properties, including:
+        - username: username of the HuggingFace user (can be an individual
+            user or an organization)
+        - access_token: access token value issued by HuggingFace for the s
+            pecified username.
+        - repo_name: the repository name to push the current version of the
+            model to. The default value is same as the TFX pipeline name.
+        - space_config: space_config carries additional values such as:
+            - app_path : path where the application templates are in the cont
+            ainer that runs the TFX pipeline. This is expressed either apps.
+            gradio.img_classifier or apps/gradio.img_classifier.
+            - repo_name : the repository name to push the application to. The
+            default value is same as the TFX pipeline name
+            - space_sdk : either gradio or streamlit. this will decide which a
+            pplication framework to be used for the Space repository. The de
+            fault value is gradio
+            - placeholders : dictionary which placeholders to replace with mod
+            el specific information. The keys represents describtions, and t
+            he values represents the actual placeholders to replace in the f
+            iles under the app_path. There are currently two predefined keys,
+            and if placeholders is set to None, the default values will be used.
+        Raises:
+            RuntimeError: if app_path is not set when space_config is provided.
+            RuntimeError: if git-lfs is not installed.
+        """
+    self._log_startup(input_dict, output_dict, exec_properties)
+
+    if self._is_git_lfs_installed() is not True:
+      raise RuntimeError(
+          "Git-LFS is not installed. "
+          "Git-LFS installation guide can be found in https://git-lfs.github.com/."
```

Comment:
nitpick: may be better to instead reference HugginFace documentation here instead.